internal/middlewares/strip_path.go

```diff
@@ -0,0 +1,22 @@
+package middlewares
+
+import (
+	"bytes"
+
+	"github.com/valyala/fasthttp"
+)
+
+// StripPathMiddleware strips the first level of a path.
+func StripPathMiddleware(next fasthttp.RequestHandler) fasthttp.RequestHandler {
```

Comment:
Why not let the reverse proxy do this with a path rewrite configuration? That would also avoid sed-ding the `href=` tags in the Dockerfiles, authelia-scripts and in server.go. What do you think?

Reply:
I think supporting base_url should means that there's minimal work required in the reverse proxy.

I cannot think of an application I utilise that supports a configurable `base_url` that also requires a rewrite, I would like to avoid this primarily from a user experience perespective but secondaryily from a support perspective.

We cannot avoid templating the `href=` tags because those can only be specified during the frontend build, I believe templating is the only way to assign this dynamically though happy to look at another implementation if you can think of one.

If we go down the rewrite path with the reverse proxy it means that users will actually need to intercept asset requests from `/static` as opposed to `/<path>/static` and that could conflict with other routes they may have setup in their reverse proxy, which really is messy because they want to share the domain in question hence the base_url modification.

Reply:
fair enough, it was just an open question.